src/EventStore.Core/TransactionLog/Chunks/TFChunkWriter.cs

```diff
@@ -40,24 +43,50 @@ public class TFChunkWriter : ITransactionFileWriter {
 			// DO NOTHING
 		}
 
+		public void Begin() {
+			if (_inTransaction)
+				throw new Exception("Attempt to begin a new transaction while there is an ongoing transaction.");
+
+			_inTransaction = true;
+		}
+
 		public bool Write(ILogRecord record, out long newPos) {
 			var result = _currentChunk.TryAppend(record);
 			if (result.Success)
-				_writerCheckpoint.Write(result.NewPosition + _currentChunk.ChunkHeader.ChunkStartPosition);
+				_nextRecordPosition = result.NewPosition + _currentChunk.ChunkHeader.ChunkStartPosition;
 			else
-				CompleteChunk(); // complete updates checkpoint internally
-			newPos = _writerCheckpoint.ReadNonFlushed();
+				CompleteChunk();
```

Comment:
```suggestion
				CompleteChunk(); // complete updates _nextRecordPosition internally
```

on more careful thought it should read like this actually

Reply:
fixed